lib/filewatch/watched_files_collection.rb

```diff
@@ -15,13 +15,16 @@ def add(watched_file)
       @sort_method.call
     end
 
-    def delete(paths)
-      Array(paths).each do |f|
-        index = @pointers.delete(f)
-        @files.delete_at(index)
-        refresh_pointers
+    def remove_paths(paths)
+      removed_files = Array(paths).map do |path|
+        if index = @pointers.delete(path)
```

Comment:
It's a matter of taste, but assignment and test should be separated for mayor intent clarity

Reply:
okay, going to change that if it's confusing. I am used to some "bad" Ruby practice :smiling_imp: